store/rootmulti/store.go

```diff
@@ -1028,9 +1028,10 @@ func (rs *Store) flushMetadata(db dbm.DB, version int64, cInfo *types.CommitInfo
 
 	flushLatestVersion(batch, version)
 
-	if err := batch.WriteSync(); err != nil {
+	if err := batch.Write(); err != nil {
```

Comment:
what is the failure and recovery flow for changing this?

Reply:
Please elaborate.

Reply:
I suppose he means what happens if the machine crashes:

```go
    // If false, and the machine crashes, then some recent writes may be lost.
    // Note that if it is just the process that crashes (and the machine does
    // not) then no writes will be lost.
```

(from goleveldb WriteOptions' Sync)

Reply:
See @yihuang's comments in the issue :)

Reply:
> what is the failure and recovery flow for changing this?

> I suppose he means what happens if the machine crashes:

When machine crashes and recent writes are lost, on startup, app latest version will be smaller than the tendermint latest version, and the recent few blocks will be replayed during ABCI handshake.

Reply:
can we write an e2e test to replicate this? It would help with any spontaneous  regressions

Reply:
since it don't happen when node restart, so it's not easy to reproduce in e2e test, the only way I can think of is directly inject the faulty code, for example implement the buffer layer and simulate the buffer lost scenario, it won't be an e2e test though.

Reply:
it would be good to have a test or some way to verify the flow and if anything changes in the flow we can catch it

Reply:
As @yihuang pointed out, it's not trivial to write such a test.

Reply:
how about:

- stop the node
- randomly deleting stuff from the end of db WAL file
- start the node

Reply:
yea that works. not sure when i will have a chance to run this test though